Edit the Feature Matrix docs page

ptgott opened the pull request:
Reflect the latest messaging guidance around Teleport products and capabilities.

ptgott committed:
Edit the Feature Matrix docs page

Reflect the latest messaging guidance around Teleport products and
capabilities.

ptgott committed:
Edit Feature Matrix formatting

- Expand the first column and narrow the final three columns.
- Use paragraphs for subsections of Teleport Zero Trust Access

roraback commented on docs/pages/feature-matrix.mdx:
```diff
@@ -13,91 +13,184 @@ ensure that all tables are aligned.*/}
-|SOC 2 Features|✔|✔|Limited|
-|[Structured Audit Logs](./reference/monitoring/audit.mdx)|✔|✔|✔|
+| [Single Sign-On](https://goteleport.com/docs/admin-guides/access-controls/sso/)| GitHub, Google Workspace, Entra, Okta, OIDC, SAML, Teleport | GitHub, Google Workspace, Entra, Okta, OIDC, SAML, Teleport | GitHub |
+| User & Group Provisioning & Deprovisioning (SCIM & Custom Protocols), including Okta and Entra | Available In Teleport Identity Governance | ✖ |
```
You're missing a cell here. We should repeat the "Available in..." text for both Cloud and Self-hosted

ptgott replied:
I'll add the list of products to the Self-Hosted column. We don't tend to italicize third-party product names in the docs, so I'm leaning toward keeping these un-italicized for consistency.

roraback commented on docs/pages/feature-matrix.mdx:
```diff
@@ -13,91 +13,184 @@ ensure that all tables are aligned.*/}
+|---|:---:|:---:|:---:|
+| Dynamic, self-updating inventory | ✔ | ✔ | ✔ |
+| Supports SSH, RDP, Kubernetes, Databases, AWS, Azure, GCP API and CLI, Web applications and services, TCP endpoints for Linux, Windows and MacOS. | ✔ | ✔ | ✔ |
+| [Machines](https://goteleport.com/docs/enroll-resources/machine-id/) & [workloads](https://goteleport.com/docs/enroll-resources/workload-identity/introduction/#teleport-workload-identity-vs-machine-id) | Available in Teleport Machine & Workload Identity |  |  |
```
We should repeat the "Available in..." text for both Cloud and Self-hosted

roraback commented on docs/pages/feature-matrix.mdx:
```diff
@@ -13,91 +13,184 @@ ensure that all tables are aligned.*/}
+|  | **Teleport Enterprise (Cloud)** | **Teleport Enterprise (Self-Hosted)** | **Teleport Community Edition** |
+|---|:---:|:---:|:---:|
+| [Role-Based Access Control](https://goteleport.com/docs/admin-guides/access-controls/guides/role-templates/) | ✔ | ✔ | ✔ |
+| [Just-in-Time Access Requests & Reviews](https://goteleport.com/docs/admin-guides/access-controls/access-requests/resource-requests/) | Available in Teleport Identity Governance |  | Only can request roles through CLI  |
```
Let's repeat the "Available in..." text for both Cloud and Self-hosted here, too

ptgott committed:
Add br tags before Feature Matrix paragraphs

This way, paragraphs that are supposed to divide the Zero Trust Access
table into subsections are easier to pick out from the tables they
divide. We can use this approach until we introduce a plugin to allow
colspans.

roraback commented:
Diana had this suggestion, which I think would be workable, and perhaps we could get a plugin added in the future that enables us to use colspans?
![image](https://github.com/user-attachments/assets/6c51bc13-e394-43f6-9b63-b14d09690d3f)

ptgott commented:
> Diana had this suggestion—a shaded row with the divisions in the left column—which I think would be workable, and perhaps we could get a plugin added in the future that enables us to use colspans?

Here's what this looks like with no shading:

<img width="734" alt="image" src="https://github.com/user-attachments/assets/2fa7e9e4-34f5-49eb-a498-3db93f7d7769" />

What color would make sense here? Maybe a very dark gray (to contrast with the light gray row shading) with white text?

ptgott committed:
Respond to roraback feedback

- Fix empty SSO cell
- Fix empty "Machines" cells
- Fix empty Access Requests cell
- Remove "Teleport" from table headers

roraback commented:
@ptgott Let's do a background of #dddddd for the entire row. It's a _bit_ visually awkward, but it also hides the borders, which is kind of nice, since we want to ensure people don't think the cells to the right are each supposed to have checkmark or an x.

<img width="923" alt="image" src="https://github.com/user-attachments/assets/5a8d7efc-c0cf-4eec-9a69-b129ea9eeaca" />

Let's also shorten all of the descriptions. Diana and I worked through this language together:
* User identity - Authenticate users without passwords:
* Resource identity - Assign a cryptographic identity to every Teleport Protected Resource:
* Secure remote access - Zero-trust, auditable access to your infrastructure:
* Short-lived privileges - Ephemeral authorization granted through short-lived certificates:
* Session recording and interactive controls - Record, replay, join, and moderate interactive sessions:
* Identity-based audit events:
* Regulatory standards and frameworks:

ptgott committed:
Include table subsections as shaded rows

Edit the Feature Matrix page to:
- Shorten table section descriptions
- Use a style tag to shade certain rows of the first and final tables

ptgott commented:
@roraback I've completed this in the latest push

roraback commented:
The product sections are looking really good. One more thing that I only just caught: would you do the same row shading trick with the platform-wide tables to combine them into a single table so that it better aligns with the final table in the Google doc?

roraback commented on docs/pages/feature-matrix.mdx:
```diff
@@ -6,98 +6,169 @@ description: Provides a comparison of features available in Teleport products.
+to applications, servers, databases, Kubernetes clusters, and other resources
+across distributed infrastructures.
+
+|  | **Teleport Enterprise (Cloud)** | **Teleport Enterprise (Self-Hosted)** | **Teleport Community Edition** |
```
We say Teleport over and over in the headings and text above the table—I think we can kill it from the table header so that it doesn't take as many lines
<img width="806" alt="image" src="https://github.com/user-attachments/assets/f684a58f-5d34-4500-80f7-5a52f4684ef5" />

roraback reviewed (changes requested):
I think it'll be good once you merge the bottom tables, fix the SSO row, and shorten the table heading cells.

ptgott commented:
@roraback I've shaded the last table in 0dc3c74e577cd87c8b21b0ad8030ee5b0fe23201 and responded to other feedback in 231a9fb5dad3c3d5746673cadf62379dd4acf536

ptgott committed:
Fix spelling

ptgott committed:
Fix absolute links in the Feature Matrix

ptgott committed:
Add missing feature matrix features

Add two items to the Identity Governance section

roraback reviewed (approved):
Looks pretty good to me. I would love to shorten a bit of language in the last table, but that isn't urgent, and it could take a bit more time to sort out, so I'm on board with merging this, and we can sort out any remaining adjustments to language later.

I'd also love to have a follow-on task to add more links to docs pages, particularly in the Identity Security and Machine and Workload Identity tables... but again, I'm on board with merging this, and we can enhance it further in a future task.
